cpp/ql/src/semmle/code/cpp/PrintAST.qll

```diff
@@ -528,6 +578,239 @@ class ArrayAggregateLiteralNode extends ExprNode {
   }
 }
 
+private string getChildAccessor(Locatable parent, Element child) {
+  shouldPrintFunction(getEnclosingFunction(parent)) and
+  (
+    exists(Stmt s | s = parent.(Stmt) |
+      namedStmtChildPredicates(s, child, result)
+      or
+      not exists(string p | namedStmtChildPredicates(s, child, p)) and
```

Comment:
I actually tried that, but it doesn't work, I guess because the predicate does not have a result?
The error message is:
"Expected a predicate with result but found 'namedExprChildPredicates(Expr expr, Element ele, string pred)', which is a predicate without result."

Reply:
I guess you can do `not namedStmtChildPredicates(s, child, _)` then?